src/ray/object_manager/object_manager.cc

```diff
@@ -66,9 +68,29 @@ void ObjectManager::HandleObjectAdded(
   ObjectID object_id = ObjectID::from_binary(object_info.object_id);
   RAY_CHECK(local_objects_.count(object_id) == 0);
   local_objects_[object_id].object_info = object_info;
-  ray::Status status =
-      object_directory_->ReportObjectAdded(object_id, client_id_, object_info);
-
+  std::vector<uint8_t> inline_object_data;
+  std::string inline_object_metadata;
+  bool inline_object_flag = false;
+
+  if (object_info.data_size <= RayConfig::instance().inline_object_max_size_bytes()) {
+    // Inline object. Store the object's data in the object's GCS entry.
+    std::vector<plasma::ObjectBuffer> object_buffers;
+    ARROW_CHECK_OK(store_client_.Get({object_id.to_plasma_id()}, -1, &object_buffers));
+    inline_object_flag = true;
+    inline_object_data.assign(object_buffers[0].data->data(),
+                              object_buffers[0].data->data() + object_info.data_size);
```

Comment:
I guess they are probably the same, but it seems safer to use `object_buffers[0].data->size()` than `object_info.data_size` (and same for the metadata below). Maybe @pcmoritz can comment?